irohad/ametsuchi/impl/postgres_ordering_service_persistent_state.hpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_POSTGRES_ORDERING_SERVICE_PERSISTENT_STATE_HPP
+#define IROHA_POSTGRES_ORDERING_SERVICE_PERSISTENT_STATE_HPP
+
+#include <pqxx/pqxx>
+#include "ametsuchi/ordering_service_persistent_state.hpp"
+#include "common/result.hpp"
+#include "logger/logger.hpp"
+#include "ametsuchi/impl/postgres_wsv_common.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+
+    /**
+     * Class implements OrderingServicePersistentState for persistent storage of
+     * Ordering Service with PostgreSQL.
+     */
+    class PostgresOrderingServicePersistentState
+        : public OrderingServicePersistentState {
+     public:
+      /**
+       * Create the instance of PostgresOrderingServicePersistentState
+       * @param postgres_options postgres connection string
+       * @return new instace of PostgresOrderingServicePersistentState
+       */
+      static expected::Result<
+          std::shared_ptr<PostgresOrderingServicePersistentState>,
+          std::string>
+      create(std::string postgres_options);
+
+      /**
+       * Constructor
+       * @param postgres_connection postgres connection object
+       * @param postgres_transaction postgres transaction object
+       */
+      explicit PostgresOrderingServicePersistentState(
+          std::unique_ptr<pqxx::lazyconnection> postgres_connection,
+          std::unique_ptr<pqxx::nontransaction> postgres_transaction);
+
+      /**
+       * Initialize storage.
+       * Create tables and fill with initial value.
+       */
+      void initStorage();
```

Comment:
What is the point to make difference between interface and implementation in terms of public API?

Reply:
I think initialisation and dropStorage() is implementation-specific